orchagent/flex_counter/flowcounterrouteorch.cpp

```diff
@@ -0,0 +1,965 @@
+            {
+                break;
+            }
+            unbindFlowCounter(route_pattern, route_pattern.vrf_id, bound_iter->first, bound_iter->second);
+            pending_iter->second.erase(bound_iter);
+            --current_bound_count;
+        }
+    }
+
+    // Remove from bound cache
+    if (iter != mBoundRouteCounters.end())
+    {
+        while(current_bound_count > route_pattern.max_match_count)
+        {
+            auto bound_iter = iter->second.begin();
+            if (bound_iter == iter->second.end())
+            {
+                break;
+            }
+
+            removeRouteFlowCounterFromDB(route_pattern.vrf_id, bound_iter->first, bound_iter->second);
+            unbindFlowCounter(route_pattern, route_pattern.vrf_id, bound_iter->first, bound_iter->second);
+            iter->second.erase(bound_iter);
+            --current_bound_count;
+        }
+    }
+}
+
+void FlowCounterRouteOrch::onRoutePatternMaxMatchCountChange(RoutePattern &route_pattern, size_t new_max_match_count)
+{
+    SWSS_LOG_ENTER();
+
+    if (route_pattern.max_match_count != new_max_match_count)
+    {
+        auto old_max_match_count = route_pattern.max_match_count;
+        route_pattern.max_match_count = new_max_match_count;
+
+        if (!isRouteFlowCounterEnabled())
+        {
+            return;
+        }
+
+        auto current_bound_count = getRouteFlowCounterSizeByPattern(route_pattern);
+        SWSS_LOG_NOTICE("Current bound route flow counter count is %zu, new limit is %zu, old limit is %zu", current_bound_count, new_max_match_count, old_max_match_count);
+        if (new_max_match_count > old_max_match_count)
+        {
+            if (current_bound_count == old_max_match_count)
+            {
+                createRouteFlowCounterByPattern(route_pattern, current_bound_count);
+            }
+        }
+        else
+        {
+            if (current_bound_count > new_max_match_count)
+            {
+                reapRouteFlowCounterByPattern(route_pattern, current_bound_count);
+            }
+        }
+    }
+}
+
+void FlowCounterRouteOrch::getRouteFlowCounterNameMapKey(sai_object_id_t vrf_id, const IpPrefix& ip_prefix, std::string &key)
+{
+    SWSS_LOG_ENTER();
+    std::ostringstream oss;
+    if (gVirtualRouterId != vrf_id)
+    {
+        auto *vrf_orch = gDirectory.get<VRFOrch*>();
+        auto vrf_name = vrf_orch->getVRFname(vrf_id);
+        if (vrf_name == "")
+        {
+            getVnetNameByVrfId(vrf_id, vrf_name);
+        }
+
+        if (vrf_name != "")
+        {
+            oss << vrf_name;
+            oss << "|";
+        }
+        else
+        {
+            // Should not happen, just in case
+            SWSS_LOG_ERROR("Failed to get VRF/VNET name for vrf id %s", sai_serialize_object_id(vrf_id).c_str());
+        }
+    }
+    oss << ip_prefix.to_string();
+    key = oss.str();
+}
+
+void FlowCounterRouteOrch::handleRouteAdd(sai_object_id_t vrf_id, const IpPrefix& ip_prefix)
+{
```

Comment:
Why is `mRouteFlowCounterSupported `not checked here as in other APIs. Please add a check